### PR TITLE
TINY-5074: Improve polaris link regex

### PR DESCRIPTION
**Pull request by JamesToohey**
Related Ticket: TINY-5074

Description of Changes:
- Reduced duplication
- Replaced 0-9 with \d
- Expanded valid scheme characters and length as per https://url.spec.whatwg.org/#url-scheme-string

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] ~Branch prefixed with `feature/` for new features (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/polaris/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## 4.0.8 - 2021-07-02
+
+### Improved
+
+- Improved detection of custom URL schemes #TINY-5074
```
Comment (lnewson) [inline modules/polaris/CHANGELOG.md:11]:
```suggestion
### Fixed
- Fixed detection of custom URL schemes #TINY-5074
```
This needs to be `Fixed` if it's going to be a patch release. We also shouldn't have the blank line.

**Review by lnewson — COMMENTED** [review]
LGTM :+1: though not approving as I was a little too involved in this.

File: modules/polaris/CHANGELOG.md (hunk 1 above)
Comment (JamesToohey) [reply modules/polaris/CHANGELOG.md:11]:
```suggestion
### Fixed
- Fixed incorrect `Regexes.link` URL detection for custom schemes #TINY-5074
```

**Review by ltrouton — APPROVED** [review]

**Review by ashfordneil — APPROVED** [review]

File: modules/polaris/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
```
Comment (JamesToohey) [inline modules/polaris/CHANGELOG.md:6]:
```suggestion

## Unreleased

```

**Review by metricjs — APPROVED** [review]

**Review by HAFRMO — APPROVED** [review]

**Pull request merged**